osbuild/pipeline.py

```diff
@@ -327,6 +327,17 @@ def run(self, store, monitor, libdir, output_directory):
         return results
 
 
+class Manifest:
+    """A Pipeline with its source options"""
+
+    def __init__(self, pipeline: Pipeline):
+        self.pipeline = pipeline
+        self.source_options = {}
```

Comment:
Can't we accept the sources as an argument here as well? If we take them in the constructor, we can at least reasonably well argue they are better not modified later on by the caller.

Reply:
Sure, good point.

Reply:
This is done, and I also renamed the member variable to `sources`, which makes more in line what the `Stage`s do and is more intuitive.